libshaderc_spvc/src/spvc.cc

```diff
@@ -338,6 +338,71 @@ shaderc_spvc_status shaderc_spvc_set_decoration(
   return status;
 }
 
+shaderc_spvc_status shaderc_spvc_get_decoration(
+    const shaderc_spvc_context_t context, uint32_t id,
+    shaderc_spvc_decoration decoration, uint32_t* argument_ptr) {
+  uint32_t& argument = *argument_ptr;
```

Comment:
You should test that argument_ptr is not null.

Reply:
Also no need to create the LHS argument reference, you can use *argument_ptr later to get and set values.

Reply:
Yeah. It's a habit intending to make the code look nicer.

Reply:
I prefer using pointers over references when possible, since the * syntax makes it clear that you writing out of the value out of scope.